hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/ModifyTableProcedure.java

```diff
@@ -512,6 +527,23 @@ private void closeExcessReplicasIfNeeded(MasterProcedureEnv env) {
       .createUnassignProceduresForClosingExcessRegionReplicas(getTableName(), newReplicaCount));
   }
 
+  private void syncErasureCodingPolicy(MasterProcedureEnv env) throws IOException {
+    String oldPolicy = unmodifiedTableDescriptor.getErasureCodingPolicy();
```

Comment:
The first 4 lines can be refactored into neeSynccErasureCodingPolicy and same can be used as if condition while deciding next state in the procedure